src/main/java/com/blogify/blogapi/service/PostFileService.java

```diff
@@ -0,0 +1,67 @@
+package com.blogify.blogapi.service;
+
+import static com.blogify.blogapi.service.utils.ExceptionMessageBuilderUtils.notFoundByIdMessageException;
+
+import com.blogify.blogapi.constant.FileConstant;
+import com.blogify.blogapi.endpoint.rest.model.PostPicture;
+import com.blogify.blogapi.file.BucketComponent;
+import com.blogify.blogapi.file.S3Service;
+import com.blogify.blogapi.model.exception.NotFoundException;
+import com.blogify.blogapi.repository.PostPictureRepository;
+import com.blogify.blogapi.repository.model.Post;
+import java.io.IOException;
+import java.util.List;
+import javax.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+@AllArgsConstructor
+public class PostFileService {
+  private final PostService postService;
+  private final PostPictureRepository repository;
+  private final S3Service s3Service;
+  private final BucketComponent bucketComponent;
```

Comment:
S3Service and BucketComponent are linked so you should call it in S3Service for more abstraction

Reply:
BucketComponent and S3Service use two different SDKs:
- BucketComponent uses s3-transfer-manager SDK: it only manages real File and Directory from the client side (poja generated, and limited), it's main purpose is to check s3 health.
- S3Service uses s3 SDK: use to upload byte[] and delete files in s3 bucket.